components/ws-manager/pkg/manager/pod_controller.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021 gitpod.io.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manager
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// PodReconciler reconciles a Pod object
+type PodReconciler struct {
```

Comment:
While reading through I was left wondering why a `_controller.go` containers a `*Reconciler` only. Is this a pattern in Kubernetes controller?

Personally, I'd probably throw the `ConfigmapReconciler` and `PodReconciler` into a `reconciler.go` - but that's nothing more than personal taste :)

Reply:
>  Is this a pattern in Kubernetes controller?

Yes, but also the _controller.go file should contain the logic/helpers in the *Reconciler method. I see this PR as the first step to refactor the code in a different PR.